docs/rule/worker_tool.soy

```diff
@@ -75,6 +75,17 @@
   {/param}
 {/call}
 
+{call buck.arg}
+  {param name: 'persistent' /}
+  {param default: 'False' /}
+  {param desc}
+    If set to true, buck will not restart the tool unless the tool itself changes. This means the
+    tool will persist across multiple buck commands without being shut down and may see the same
+    rule being built more than once. Be careful when using this with tools that don't expect
+    to process the same input (with different contents) twice!
+  {/param}
+{/call}
+
```

Comment:
Ah I see, I didn't realize that. I think from the perspective of the code, it's still one build though - each time a `BuildCommand` gets created a new `ExecutionContext` gets created with it (which is what holds the ephemeral workers right now).

The issue I'm facing is with the Clojure compiler (though I think Scala may have a similar issue). The basic problem is that loading files - compiled or not - is really slow because it executes a lot of `static`-level code. A good way to "cache" that load step is simply to not unload the output. The current `worker_tool` is perfect for that. Let's say I have 2 rules with dependencies: `//:a -> //:b`.
 - On a clean build of `//:a`, a new `worker_tool` simply compiles `//:b` and keeps it in-memory. Then when `//:a` is compiled, we add the output of `//:b` to the classpath, but the classloader ignores it since the compiled class is already loaded.
 - If we change `//:a`'s input and build `//:a` again, the new `worker_tool` will add `//:b`'s output to the classpath to load the compiled code rather than recompiling `//:b` again.
 - Unfortunately this doesn't work if a new "version" of `//:b` is compiled with the same worker - there's really no way to force unload a class in java cleanly.

With this approach the `worker_tool` never ends up reloading code unless it really needs to. The downside is that targets need precise deps or the builds won't be deterministic. The way that's accomplished in the project I linked above is via an autodeps-like mechanism that reads the code to figure out the exact dependency graph.

Is there a chance we can keep the two kinds of persistent processes coexisting (though perhaps not as a default)?